### PR TITLE
Email with long value is slowing down and crashing

### PR DESCRIPTION
**Pull request by LouiaR**
Fixes https://github.com/comicrelief/giftaid-react/issues/250

### CODE REVIEW

**Comment by AndyEPhipps** [comment]
@LouiaR can you explain what this change is doing please?

**Comment by LouiaR** [comment]
> @LouiaR can you explain what this change is doing please?

I change the regex used for email validation, as when the email is too long like `agdbdreyrugjgghgkadqewyeorpfkhi@jruqpalkdidhhagtryhfhf ` the typing is slowed if I user still keep continuing entering more chars then suddenly keys are frozen.

**Comment by AndyEPhipps** [comment]
> > @LouiaR can you explain what this change is doing please?
> 
> I change the regex used for email validation, as when the email is too long like `agdbdreyrugjgghgkadqewyeorpfkhi@jruqpalkdidhhagtryhfhf ` the typing is slowed, if user still keep continuing entering more chars then suddenly keys are frozen.

Nah I mean I understand the problem, but why does changing the regex fix it?

**Comment by LouiaR** [comment]
> > > @LouiaR can you explain what this change is doing please?
> > 
> > 
> > I change the regex used for email validation, as when the email is too long like `agdbdreyrugjgghgkadqewyeorpfkhi@jruqpalkdidhhagtryhfhf ` the typing is slowed, if user still keep continuing entering more chars then suddenly keys are frozen.
> 
> Nah I mean I understand the problem, but why does changing the regex fix it?

It stop slowing down effect when typing and keys being in frozen state

**Comment by AndyEPhipps** [comment]
> > > > @LouiaR can you explain what this change is doing please?
> > > 
> > > 
> > > I change the regex used for email validation, as when the email is too long like `agdbdreyrugjgghgkadqewyeorpfkhi@jruqpalkdidhhagtryhfhf ` the typing is slowed, if user still keep continuing entering more chars then suddenly keys are frozen.
> > 
> > 
> > Nah I mean I understand the problem, but why does changing the regex fix it?
> 
> It stops slowing down effect when typing and keys being in frozen state

**Comment by AndyEPhipps** [comment]
whoops

File: src/components/InputField/validation.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 const defaultValidationPatterns = {
   tel: /^[0-9 ]{11,}$/,
   number: /^[0-9]+$/,
-  email: /^([A-Za-z0-9_+-]+\.?)*[A-Za-z0-9_+-]+@[A-Za-z0-9]+([A-Za-z0-9_-]+\.?)*[A-Za-z0-9]+\.[A-Za-z]{2,}$/,
+  // eslint-disable-next-line no-useless-escape
```
Comment (AndyEPhipps) [inline src/components/InputField/validation.js:4]: Were you not able to get around this? Not a big deal, but we're gonna have to use this pattern in multiple projects; would be nice to get it clean

Comment (LouiaR) [reply src/components/InputField/validation.js]: Oops forgot this
Done

Comment (AndyEPhipps) [reply src/components/InputField/validation.js]: Thanks man, will doublecheck once the test env has rebuilt

**Pull request merged**